Formula/v8.rb

```diff
@@ -15,98 +15,89 @@ class V8 < Formula
-    (buildpath/"tools/swarming_client").install resource("swarming_client")
-    (buildpath/"testing/gtest").install resource("gtest")
-    (buildpath/"testing/gmock").install resource("gmock")
-    (buildpath/"tools/clang").install resource("clang")
-
-    system "make", "native", "library=shared", "snapshot=on",
-                   "console=readline", "i18nsupport=off",
-                   "strictaliasing=off"
-
-    include.install Dir["include/*"]
-
-    cd "out/native" do
-      rm ["libgmock.a", "libgtest.a"]
-      lib.install Dir["lib*"]
-      bin.install "d8", "mksnapshot", "process", "shell" => "v8"
+    # Add depot_tools in PATH
+    ENV.prepend_path "PATH", buildpath
+    # Prevent from updating depot_tools on every call
+    # see https://www.chromium.org/developers/how-tos/depottools#TOC-Disabling-auto-update
+    ENV["DEPOT_TOOLS_UPDATE"] = "0"
+
+    # Initialize and sync gclient
+    system "gclient", "root"
+    system "gclient", "config", "--spec", <<~EOS
+      solutions = [
+        {
+          "url": "https://chromium.googlesource.com/v8/v8.git",
+          "managed": False,
+          "name": "v8",
+          "deps_file": "DEPS",
+          "custom_deps": {},
+        },
+      ]
+      target_os = [ "mac" ]
+      target_os_only = True
+      cache_dir = "#{HOMEBREW_CACHE}/gclient_cache"
+    EOS
+
+    system "gclient", "sync",
+      "-j", ENV.make_jobs,
+      "-r", version,
+      "--no-history",
+      "-vvv"
+
+    # Enter the v8 checkout
+    cd "v8" do
+      output_path = "out.gn/x64.release"
+
+      gn_args = {
+        :is_debug => false,
+        :is_component_build => true,
+        :v8_use_external_startup_data => false,
+        :v8_enable_i18n_support => true,
+      }
+
+      # Transform to args string
+      gn_args_string = gn_args.map { |k, v| "#{k}=#{v}" }.join(" ")
+
+      # Build with gn + ninja
+      system "gn",
+        "gen",
+        "--args=#{gn_args_string}",
+        output_path
+
+      system "ninja",
+        "-j", ENV.make_jobs,
+        "-C", output_path,
+        "-v",
+        "d8"
+
+      # Install all the things
+      include.install Dir["include/*"]
+
+      cd output_path do
+        lib.install Dir["lib*.dylib"]
+
+        # Install d8 and icudtl.dat in libexec and symlink
+        # because they need to be in the same directory.
+        libexec.install Dir["d8", "icudt*.dat"]
+        (bin/"d8").write <<~EOS
+          #!/bin/bash
+          exec "#{libexec}/d8" --icu-data-file="#{libexec}/icudtl.dat" "$@"
+        EOS
+      end
     end
   end
 
   test do
-    assert_equal "Hello World!", pipe_output("#{bin}/v8 -e 'print(\"Hello World!\")'").chomp
+    assert_equal "Hello World!", shell_output("#{bin}/d8 -e 'print(\"Hello World!\");'").chomp
+    assert_match %r{12/\d{2}/2012}, shell_output("#{bin}/d8 -e 'print(new Intl.DateTimeFormat(\"en-US\").format(new Date(Date.UTC(2012, 11, 20, 3, 0, 0))));'").chomp
```

Comment:
Suppose there's absolutely no chance we can get this anywhere near 80 characters?